pybossa/api/__init__.py

```diff
@@ -384,26 +384,23 @@ def task_progress(project_id=None, short_name=None):
         project = project_repo.get_by_shortname(short_name)
     elif project_id:
         project = project_repo.get(project_id)
-
     filter_fields = request.args
-
     if not project:
         return abort(404)
 
     sql_text = "SELECT COUNT(*) FROM task WHERE project_id=" + str(project.id)
-
     task_info_fields = get_searchable_columns(project.id)
 
     # create sql query from filter fields received on request.args
     for key in filter_fields.keys():
         if key in task_fields:
             sql_text += " AND {0}=:{1}".format(key, key)
+        elif key in task_info_fields and filter_fields[key] == "null".lower():
```

Comment:
This could possibly be optimized as below. Also `filter_fields[key].lower()` rather than "null".lower(). Thanks!

```python
elif key in task_info_fields:
  if filter_fields[key].lower() == "null":
    sql_text +=  " AND info ->> '{0}' is Null".format(key)    
  else:
    sql_text += " AND info ->> '{0}'=:{1}".format(key, key)
```